app/components/autocomplete/slash_suggestion/app_slash_suggestion/app_slash_suggestion.tsx

```diff
@@ -0,0 +1,238 @@
+            return;
+        }
+
+        if (!this.isAppCommand(nextValue, this.props.channelId, this.props.rootId)) {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        this.fetchAndShowAppCommandSuggestions(nextValue, this.props.channelId, this.props.currentTeamId, this.props.rootId);
+    }
+
+    isAppCommand = (pretext: string, channelID: string, teamID = '', rootID?: string) => {
+        this.appCommandParser.setChannelContext(channelID, teamID, rootID);
+        return this.appCommandParser.isAppCommand(pretext);
+    }
+
+    fetchAndShowAppCommandSuggestions = async (pretext: string, channelID: string, teamID = '', rootID?: string) => {
+        this.appCommandParser.setChannelContext(channelID, teamID, rootID);
+        const suggestions = await this.appCommandParser.getSuggestions(pretext);
+        this.updateSuggestions(suggestions);
+    }
+
+    updateSuggestions = (matches: ExtendedAutocompleteSuggestion[]) => {
+        this.setState({
+            active: Boolean(matches.length),
+            dataSource: matches,
+        });
+        this.props.onResultCountChange(matches.length);
+    }
+
+    completeSuggestion = (command: string) => {
+        const {onChangeText} = this.props;
+        analytics.trackCommand('complete_suggestion', `/${command} `);
+
+        // We are going to set a double / on iOS to prevent the auto correct from taking over and replacing it
+        // with the wrong value, this is a hack but I could not found another way to solve it
+        let completedDraft = `/${command} `;
+        if (Platform.OS === 'ios') {
+            completedDraft = `//${command} `;
+        }
+
+        onChangeText(completedDraft);
+
+        if (Platform.OS === 'ios') {
+            // This is the second part of the hack were we replace the double / with just one
+            // after the auto correct vanished
+            setTimeout(() => {
+                onChangeText(completedDraft.replace(`//${command} `, `/${command} `));
+            });
+        }
+
+        if (!isMinimumServerVersion(Client4.getServerVersion(), 5, 24)) {
+            this.setState({
+                active: false,
+            });
+        }
+    };
+
+    completeUserSuggestion = (base: string): (username: string) => void => {
+        return () => {
+            this.completeSuggestion(base);
+        };
+    }
+
+    completeChannelMention = (base: string): (channelName: string) => void => {
+        return () => {
+            this.completeSuggestion(base);
+        };
+    }
+
+    keyExtractor = (item: ExtendedAutocompleteSuggestion): string => item.Suggestion + item.type + item.item;
+
+    renderItem = ({item}: {item: ExtendedAutocompleteSuggestion}) => {
+        switch (item.type) {
+        case COMMAND_SUGGESTION_USER:
+            return (
+                <AtMentionItem
+                    testID={`autocomplete.at_mention.item.${item.item}`}
```

Comment:
Any reason why this has a `testID` and the others don't?

Reply:
Copy paste. The ChannelMention used to have also, but typescript was not properly recognizing the types, so I removed it before finding even more errors and migrating it completely. I can put it back, or remove both.

Reply:
Do add `TestID` specially if you have an E2E test. if there is no E2E test do consider adding one.

cc: @josephbaylon